gpiocontrol/source/esp8266/CMakeLists.txt

```diff
@@ -45,45 +57,39 @@ set(ESP8266_LIBDIR "-L${CMAKE_CURRENT_SOURCE_DIR}/build -L${ESP_TOOLCHAIN_DIR}/.
                     
 set(ESP8266_LIBDIR "${ESP8266_LIBDIR} -L${ESP_SDK_BASE}/lib -L${CMAKE-CMAKE_CURRENT_SOURCE_DIR} -L${ESP_TOOLCHAIN_DIR}/ld")
 
-set(ESP8266_LDFLAGS "${ESP8266_LIBDIR} -T${ESP8266_LINKSCRIPT} -nostdlib -u call_user_start -static --start-group ${ESP8266_LIBS} -lesp-platform -lkaac_s -l${APP_NAME}_s --end-group --gc-sections -Map map")
+set(ESP8266_LDFLAGS "${ESP8266_LIBDIR} -T${ESP8266_LINKSCRIPT} -nostdlib -u call_user_start -static --start-group ${ESP8266_LIBS} -lesp-platform -lkaac_s --end-group --gc-sections -Map map")
 
 set(CMAKE_AR ${ESP_TOOLCHAIN_DIR}/xtensa-lx106-elf-ar)
 
 set(CMAKE_RANLIB ${ESP_TOOLCHAIN_DIR}/xtensa-lx106-elf-ranlib)
 
-set(CMAKE_C_LINK_EXECUTABLE "${ESP_TOOLCHAIN_DIR}/xtensa-lx106-elf-ld <LINK_FLAGS> <OBJECTS>")
-
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DESP8266_PLATFORM -Os -Wpointer-arith -Wl,-EL -fno-inline-functions -nostdlib -mlongcalls -mtext-section-literals -D__ets__ -DICACHE_FLASH")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DSSID='\"${SSID}\"' -DPWD='\"${PWD}\"' -DDEMO_ACCESS_TOKEN='\"${DEMO_ACCESS_TOKEN}\"'")
+
+set(CMAKE_C_LINK_FLAGS "${ESP8266_LDFLAGS}")
+set(CMAKE_C_LINKER "${ESP_TOOLCHAIN_DIR}/xtensa-lx106-elf-ld")
+set(CMAKE_C_LINK_EXECUTABLE "${CMAKE_C_LINKER} <OBJECTS> <CMAKE_C_LINK_FLAGS> <LINK_LIBRARIES> -o <TARGET>")
```

Comment:
This should be addressed in the scope of http://jira.kaaproject.org/browse/APP-88